### ISSUE TITLE
cWorld:TryGetHeight(x,z) deadlocks(?)

### ISSUE THREAD

**Issue by warmist** [mc-server/MCServer#2226]
when trying to create a plugin i was calling TryGetHeight with randomized coordinates (centered on player). Sometimes it would crash/hang or deadlock.

Example code:

    function hang(player )
        local center=player:GetPosition()
        local w=player:GetWorld()
        for i=1,10 do
            local distance=math.random()*250
            local angle=math.random()*math.pi
            local x=center.x+math.cos(angle)*distance
            local z=center.z+math.sin(angle)*distance
            print(center.x,center.z,x,z)
            local ok,h=w:TryGetHeight(x,z)
            print(ok,h,center.y)
        end
    end

When called every few seconds (with cRoot:Get():ForEachPlayer(hang))

**Comment by warmist** [comment]
Now it deadlocked/crashed without that, so not sure if it's something else (e.g. calling stuff from tick callback?)

**Comment by worktycho** [comment]
What OS is this running on? Because it might be a problem with recursive mutexs.

**Comment by warmist** [comment]
Yeah looks like it's a deadlock loop: e.g. GetBlock with chunk loading item both lock m_CSLayers...
Will debug more later...

**Comment by warmist** [comment]
Windows 7 x64 with x64 server.

**Comment by madmaxoft** [comment]
If it deadlocks / crashes, it should produce a crashdump (*.dmp); is there any such thing in the executable folder?

**Comment by warmist** [comment]
After working around it: it looks like the "ForEachPlayer" deadlocks with block manipulation/reading. And it did not produce dumps.
